File: julia/basics/word_frequency.jl
# Count the words of a fixed text, most frequent first and alphabetically within a tie.
# Run: julia word_frequency.jl

const TEXT = "the quick brown fox jumps over the lazy dog the fox barks"

counts = Dict{String,Int}()
for word in split(TEXT)
    counts[word] = get(counts, word, 0) + 1
end

for (word, count) in sort(collect(counts), by = pair -> (-pair.second, pair.first))
    println("$word $count")
end

File: swift/basics/word_frequency.swift
// Count the words of a fixed text, most frequent first and alphabetically within a tie.
// Run: swift word_frequency.swift

let text = "the quick brown fox jumps over the lazy dog the fox barks"

var counts: [String: Int] = [:]
for word in text.split(separator: " ") {
    counts[String(word), default: 0] += 1
}

let ranked = counts.sorted { left, right in
    left.value == right.value ? left.key < right.key : left.value > right.value
}
for (word, count) in ranked {
    print("\(word) \(count)")
}
